libprecompiled/ChainGovernancePrecompiled.cpp

```diff
@@ -70,158 +75,499 @@
+            result = verifyAndRecord(_context, Operation::UpdateThreshold,
+                CGP_UPDATE_AUTH_THRESHOLD, boost::lexical_cast<string>(threshold), _origin.hex());
+            CHAIN_GOVERNANCE_LOG(INFO)
+                << LOG_DESC("updateThreshold") << LOG_KV("origin", _origin.hex())
+                << LOG_KV("threshold", weight) << LOG_KV("return", result);
+        }
+        getErrorCodeOut(callResult->mutableExecResult(), result);
+    }
+    else if (func == name2Selector[CGP_METHOD_LIST_CM])
+    {  // listCommitteeMembers();
+        Address contractAddress;
+        abi.abiOut(data, contractAddress);
+        auto resultJson = queryCommitteeMembers(_context, SYS_ACCESS_TABLE);
+        callResult->setExecResult(abi.abiIn("", resultJson));
+    }
+    else if (func == name2Selector[CGP_METHOD_GRANT_OP])
+    {  // grantOperator(address)
+        Address user;
+        abi.abiOut(data, user);
+        result = grantOperator(_context, user.hex(), _origin);
+        getErrorCodeOut(callResult->mutableExecResult(), result);
+    }
+    else if (func == name2Selector[CGP_METHOD_REVOKE_OP])
+    {  // revokeOperator(address)
+        Address user;
+        abi.abiOut(data, user);
+        auto userAddress = user.hex();
+        do
+        {
+            Table::Ptr acTable = openTable(_context, SYS_ACCESS_TABLE);
+            auto condition = acTable->newCondition();
+            condition->EQ(SYS_AC_ADDRESS, userAddress);
+            auto entries = acTable->select(SYS_TABLES, condition);
+            if (entries->size() == 0u)
+            {
+                result = CODE_OPERATOR_NOT_EXIST;
+                CHAIN_GOVERNANCE_LOG(INFO)
+                    << LOG_DESC("revokeOperator operator not exist")
+                    << LOG_KV("operator", userAddress) << LOG_KV("return", result);
+                break;
+            }
+            result = acTable->remove(SYS_TABLES, condition, make_shared<AccessOptions>(_origin));
+            if (result == storage::CODE_NO_AUTHORIZED)
+            {
+                CHAIN_GOVERNANCE_LOG(INFO)
+                    << LOG_DESC("permission denied") << LOG_KV("operator", userAddress)
+                    << LOG_KV("origin", _origin.hex());
+                BOOST_THROW_EXCEPTION(
+                    PrecompiledException("Permission denied. " + _origin.hex() +
+                                         " can't grantOperator operator " + userAddress));
+                break;
+            }
+            CHAIN_GOVERNANCE_LOG(INFO)
+                << LOG_DESC("revoke Member") << LOG_KV("origin", _origin.hex())
+                << LOG_KV("operator", userAddress) << LOG_KV("return", result);
+        } while (0);
+
+        getErrorCodeOut(callResult->mutableExecResult(), result);
+    }
+    else if (func == name2Selector[CGP_METHOD_LIST_OP])
+    {  // listOperators()
+        Address contractAddress;
+        abi.abiOut(data, contractAddress);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'tableName' is assigned a value that is never used.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5261935)